### PR TITLE
quantlib: init at 1.29

### PR DESCRIPTION
**Pull request by ghost**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/quantlib/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ boost ];
+
+  passthru.updateScript = gitUpdater { };
+
+  meta = with lib; {
+    description = "A free/open-source library for quantitative finance";
+    homepage = "https://quantlib.org";
+    platforms = platforms.unix;
+    license = licenses.quantlib;
```
Comment (figsoda) [inline pkgs/development/libraries/quantlib/default.nix:29]: Not entirely sure about how licensing works, are we allowed to just use bsd3 here? Also the license page links to the bsd2 license but includes the text of bsd3 license for some reason

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -35256,6 +35256,8 @@ with pkgs;
 
   quantumminigolf = callPackage ../games/quantumminigolf { };
 
+  quantlib = callPackage ../development/libraries/quantlib { };
```
Comment (figsoda) [inline pkgs/top-level/all-packages.nix:35259]: please put this under `### DEVELOPMENT / LIBRARIES`

File: lib/licenses.nix [hunk 3]
```diff
@@ -828,6 +828,11 @@ in mkLicense lset) ({
     fullName = "Qhull License";
   };
 
+  quantlib = {
```
Comment (AndersonTorres) [inline lib/licenses.nix:831]: You can use `bsd2Clause` instead.

File: pkgs/development/libraries/quantlib/default.nix (hunk 1 above)
Comment (AndersonTorres) [inline pkgs/development/libraries/quantlib/default.nix:29]:
```suggestion
    license = licenses.bsd3;
```

File: pkgs/development/libraries/quantlib/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    owner = "lballabio";
+    repo = "QuantLib";
+    rev = "QuantLib-v${version}";
+    sha256 = "sha256-TpVn3zPru/GtdNqDH45YdOkm7fkJzv/qay9SY3J6Jiw=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ boost ];
+
+  passthru.updateScript = gitUpdater { };
```
Comment (AndersonTorres) [inline pkgs/development/libraries/quantlib/default.nix:23]: Unneeded because automatic.
```suggestion
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/quantlib/default.nix (hunk 4 above)
Comment (candyc1oud) [reply pkgs/development/libraries/quantlib/default.nix]: I fixed it 
thanks

File: pkgs/development/libraries/quantlib/default.nix (hunk 1 above)
Comment (candyc1oud) [reply pkgs/development/libraries/quantlib/default.nix]: ok I fikex it

**Review by AndersonTorres — APPROVED** [review]

**Comment by figsoda** [comment]
it seems to include some test files under `$out/bin`, is that intentional?
also, I'm not familiar with packaging of C libraries, should we split the outputs to stuff like `dev` and `lib`?

**Comment by candyc1oud** [comment]
> it seems to include some test files under `$out/bin`, is that intentional?
> also, I'm not familiar with packaging of C libraries, should we split the outputs to stuff like `dev` and `lib`?

I see that many distributions are packaged with files in `bin` directory

**Comment by candyc1oud** [comment]
> The multi-output split is a good idea

I finished it

File: pkgs/development/libraries/quantlib/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+, fetchFromGitHub
+, cmake
+, boost
+}:
+
+stdenv.mkDerivation rec {
+  pname = "quantlib";
+  version = "1.29";
+
+  outputs = ["lib" "dev" "out" "bin"];
```
Comment (figsoda) [inline pkgs/development/libraries/quantlib/default.nix:12]:
```suggestion
  outputs = [ "dev" "lib" "out" ];
```

> error: builder for '/nix/store/bmn5nknwvaf0n4cjj69bxkl637cf5sqm-quantlib-1.29.drv' failed to produce output path for output 'bin' at '/nix/store/bmn5nknwvaf0n4cjj69bxkl637cf5s
qm-quantlib-1.29.drv.chroot/nix/store/03ckjcp24qa64fg6k1jjaywi79920ghl-quantlib-1.29-bin'

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/quantlib/default.nix (hunk 5 above)
Comment (candyc1oud) [reply pkgs/development/libraries/quantlib/default.nix]: I was in a bit of a hurry, I pushed it without checking, I had already fixed it

Comment (candyc1oud) [reply pkgs/development/libraries/quantlib/default.nix]: I was in a bit of a hurry, I pushed it without checking, I had already fixed it
By the way, the code you gave will fail on my machine

Comment (figsoda) [reply pkgs/development/libraries/quantlib/default.nix]: I think you forgot `lib`

Comment (figsoda) [reply pkgs/development/libraries/quantlib/default.nix]: sorry, my bad, I didn't work, you can remove `lib` 🤦

Comment (candyc1oud) [reply pkgs/development/libraries/quantlib/default.nix]: ok I removed it

**Review by figsoda — APPROVED** [review]

**Pull request merged**